node/src/handlers/get-file-set-download.js

```diff
@@ -51,7 +82,102 @@ function downloadAvailable(doc) {
   );
 }
 
-async function processDownload(doc, email) {
+function isImageDownload(doc) {
+  return (
+    doc.found &&
+    ["Access", "Auxiliary"].includes(doc._source.role) &&
+    doc._source.mime_type != null &&
+    ["image"].includes(doc._source.mime_type.split("/")[0])
+  );
+}
+
+function derivativeKey(doc) {
+  const id = doc._id;
+  let prefix =
+    id.slice(0, 2) +
+    "/" +
+    id.slice(2, 4) +
+    "/" +
+    id.slice(4, 6) +
+    "/" +
+    id.slice(6, 8);
+  return "derivatives/" + prefix + "/" + id;
+}
+
+function extensionFromMimeType(mimeType) {
```

Comment:
I think this implementation works fine for now, but I'd be more comfortable with the [`mime.extension(type)`](https://www.npmjs.com/package/mime-types#mimeextensiontype) method from the mime-types package. That's not currently a top-level project dependency, but it's already pulled in indirectly by `axios`, so we wouldn't be adding bloat by making use of it. And it may prevent other issues down the road.